crates/debugging/src/trace/collect.rs

```diff
@@ -105,3 +109,30 @@ fn collect_transformed_calldata(call_trace: &CallTrace, abi: &[AbiEntry]) -> Tra
         .expect("calldata should be successfully transformed"),
     )
 }
+
+fn collect_call_result(call_trace: &CallTrace, abi: &[AbiEntry]) -> CallResult {
+    CallResult(match &call_trace.result {
+        CheatnetCallResult::Success { ret_data } => {
+            let ret_data = reverse_transform_output(
+                ret_data,
+                abi,
+                &call_trace.entry_point.entry_point_selector.0,
+            )
+            .expect("call result should be successfully transformed");
+            format_result_message("success", &ret_data)
+        }
+        CheatnetCallResult::Failure(failure) => match failure {
+            CallFailure::Panic { panic_data } => {
+                format_result_message("panic", &format_panic_data(panic_data))
+            }
+            CallFailure::Error { msg } => format_result_message("error", &msg.to_string()),
+        },
+    })
+}
+
+fn format_result_message(tag: &str, message: &str) -> String {
+    message
+        .is_empty()
+        .then(|| tag.to_string())
+        .unwrap_or_else(|| format!("{tag}: {message}"))
```

Comment:
Wouldn't this be clearer?

```suggestion
    if message.is_empty() {
        tag.to_string()
    } else {
        format!("{tag}: {message}")
    }
```

Reply:
I would say that "clean code" is more a matter of preference here. I really like the functional style, so I went with the current solution. If you prefer the imperative style, we can go with that instead — no biggie for me.

Reply:
I like functional style, but your solution seems bit too smart for what it's doing.